src/mesh/meshblock.cpp

```diff
@@ -179,25 +179,25 @@ void MeshBlock::Initialize(int igid, int ilid, LogicalLocation iloc,
   // potentially used in the load balancing calculation, but not all
   // variables that we may want to communicate are necessarily relevant
   // to the cost per meshblock.
-  const auto vars = real_container
-                        ->GetVariablesByFlag({Metadata::Independent, Metadata::FillGhost,
-                                              Metadata::ForceRemeshComm},
-                                             false)
-                        .vars();
-  for (int n = 0; n < vars.size(); ++n) {
-    RegisterMeshBlockData(vars[n]);
+  using FS_t = Metadata::FlagCollection;
+  // union of Independent and FillGhost
+  FS_t flags({Metadata::Independent, Metadata::FillGhost}, true);
+  // Toss in RemeshComm for this one
+  const auto vars =
+      real_container->GetVariablesByFlag(flags + FS_t({Metadata::ForceRemeshComm}, true))
+          .vars();
+  for (const auto &v : vars) {
+    RegisterMeshBlockData(v);
   }
 
+  // No RemeshComm
   if (pm->multilevel) {
-    const auto refine_vars =
-        real_container
-            ->GetVariablesByFlag({Metadata::Independent, Metadata::FillGhost}, false)
```

Comment:
Did the meaning of the `true/false` change between `GetVariablesByFlag({Metadata::Independent, Metadata::FillGhost}, false)` and the `flags({Metadata::Independent, Metadata::FillGhost}, true);` used from above?

Reply:
Yes---I wanted the `FlagCollection` object to have the previous behavior as `GetVariablesByFlag` by default, which meant by default it's a set intersection.

But I also wanted the default behaviour of the constructor to be `true` so the optional bool in the `FlagCollection` constructor is `DoUnion` instead of `DoIntersection`. If this is too confusing, I can change the name/defaults so the meaning of `true/false` is retained.